### PR TITLE
reports.html: Create dynamic report listing by year

### PR DESCRIPTION
**Pull request by samuelpiltch**
Replace hard coded year headings with auto-generated year headings based on reports in `_reports`.

Regards issue #670

### CODE REVIEW

File: reports.html [hunk 1]
```diff
@@ -2,26 +2,23 @@
...
-  {% for report in site.reports %}
-    {% capture year %}{{ report.date | date: '%Y' }}{% endcapture %}
-    {% if year == '2017' %}
-      <li><a href="{{ report.url }}.html">{{ report.student }}</a>
-    {% endif %}
-  {% endfor %}
-</ul>
+    <body class='report'>
+        <h1>{{ page.title }}</h1>
+        {% assign reportsByYear = site.reports | group_by_exp:"post", "post.date | date: '%Y'" %}
```
Comment (srivama) [inline reports.html]: Shouldn't it be `group_by_exp:"report"` and `report.date`?

Comment (samuelpiltch) [reply reports.html]: @srivama Fixed

**Comment by samuelpiltch** [comment]
@srivama Rebased and changed commit message. Anything else to edit before this gets merged?

**Comment by srivama** [comment]
LGTM :+1: One last thing, I wonder if those `html`, `head` and `body` tags are required here, not sure if the code is being ingested somewhere in the body of another file. @jayvdb PTAL :)
@samuelpiltch Don't make any further changes until someone else reviews it.

**Comment by samuelpiltch** [comment]
@srivama Those should be fine since it's not using a template and I can't find 'reports.html' referenced on any other page. Added the tags so that someone can go and style the page. I also suggest that the page is linked on the index page or somewhere so that it can be found publicly after its styled.
Thanks for your help, you've been super easy to work with :)

File: reports.html [hunk 2]
```diff
@@ -2,26 +2,24 @@
...
-
-<ul>
-  {% for report in site.reports %}
-    {% capture year %}{{ report.date | date: '%Y' }}{% endcapture %}
-    {% if year == '2017' %}
-      <li><a href="{{ report.url }}.html">{{ report.student }}</a>
-    {% endif %}
-  {% endfor %}
-</ul>
+    <body class="report">
```
Comment (jayvdb) [inline reports.html]: as you can see we are using two space indents.

please always retain the style that you find in a file, if it is consistent.  The maintainers of that file probably like the style they have chosen, and wont appreciate it changing for not reason.

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/680/commits/0b69583ce1cf2eff35e6bea33ce7fd83a31a4d87

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**